sound/soc/sof/intel/hda-trace.c

```diff
@@ -74,7 +74,19 @@ int hda_dsp_trace_release(struct snd_sof_dev *sdev)
 	struct hdac_stream *hstream;
 
 	if (hda->dtrace_stream) {
+		u32 mask;
+		int ret;
+
 		hstream = &hda->dtrace_stream->hstream;
+		mask = 0x1 << hstream->index;
+
+		ret = hda_dsp_stream_reset(sdev, hstream);
+		if (ret < 0)
+			return ret;
+
+		/* couple host and link DMA if link DMA channel is idle */
+		snd_sof_dsp_update_bits(sdev, HDA_DSP_PP_BAR,	SOF_HDA_REG_PP_PPCTL, mask, 0);
+
```

Comment:
The commit touches generic code, so the "recommended programming sequence" seems wrong.